### DISCUSSION TITLE
Clone plugins in parallel

### DISCUSSION THREAD

**Discussion by jooola** [marlonrichert/zsh-snap#33] [Q&A] [answered]
I'm not sure, but it seems the first plugins installation (git clone) is not done in parallel.

Any chance to silence the output of git if not error occur and run those clones in parallel ?

**Comment by marlonrichert** [comment]
@jooola I'm not sure what you're talking about. Can you give me a minimal test case with which I can reproduce the problem?

**Comment by jooola** [comment]
Here is the output when I open the first time zsh with my znap configuration:
![image](https://user-images.githubusercontent.com/19195485/105590857-77b6a900-5d92-11eb-82ba-dbd18049b958.png)

It is really verbose (I don't really mind but could be prettier if no error occur), and by the look of it, doesn't seem to run in parallel. Cloning seems to be processed one after the other.

But again, I m not sure if that is really the case. Maybe stdout doesn't reflect the actual process behind.
I can upload a asciinema if you like.

Here is my znap setup:
```sh
[...]

zstyle ':znap:*' plugins-dir ~/.zsh/
source ~/.zsh/znap/znap.zsh

znap source zsh-users/zsh-completions zsh-completions.plugin.zsh
znap source zsh-users/zsh-autosuggestions zsh-autosuggestions.zsh
znap source zsh-users/zsh-history-substring-search
znap source zsh-users/zsh-syntax-highlighting
znap source starcraftman/zsh-git-prompt

if [[ $IS_DESKTOP -eq 1 ]]; then
  znap source greymd/tmux-xpanes
fi

[...]
```

**Comment by marlonrichert** [comment, marked as answer]
Right now, Znap only clones in parallel if you put all your plugins into one `znap clone` statement. So, if you want that, you need to put this before your `znap source` statements:
```zsh
znap clone zsh-users/zsh-completions \
  zsh-users/zsh-autosuggestions \
  zsh-users/zsh-history-substring-search \
  zsh-users/zsh-syntax-highlighting \
  starcraftman/zsh-git-prompt \
  greymd/tmux-xpanes
```

Personally, I prefer to see everything that’s happening, but I could introduce an `-s` flag for silent mode.

I could also look into adding an “async clone” option for `znap source` or some “smart” logic for doing this automatically, but one has to be careful with stuff like that. If statements in your `.zshrc` get executed in random order, unexpected things can happen.

**Comment by jooola** [comment]
My bad, haven't RTFM properly...

Thanks for the help. I'll try to move from zplug to znap. I don't have much reason to have anything more complex.
